indexers/keyvalue/RedisDBIndexer/__init__.py

```diff
@@ -69,9 +67,32 @@ def query(self, key: int, *args, **kwargs) -> Optional[bytes]:
         with self.get_add_handler() as redis_handler:
             for _key in redis_handler.scan_iter(match=key):
                 res = {
-                  "key": _key,
-                  "values": redis_handler.get(_key),
+                    "key": _key,
+                    "values": redis_handler.get(_key),
                 }
                 result.append(res)
 
-        return result
\ No newline at end of file
+        return result
+
+    def update(self, keys: Iterator[int], values: Iterator[bytes], *args, **kwargs):
+        """updates the keys if they exist
+        """
+        missed = []
+        for key in keys:
+            if len(self.query(key)) == 0:
+                missed.append(key)
+        if missed:
+            raise KeyError(f'Key(s) {missed} were not found in redis')
+
+        # hack
+        self.query_handler.close()
+        self.handler_mutex = False
```

Comment:
Sorry for only asking that now, but can't we use the same handler for query and index with redis? In the end it is just something like the redis client.

Reply:
And as far as I know, the redis client can do both in parallel without problems.

Reply:
Hah even easier, just always use the same handler.

Reply:
Nice catch - I removed the hack